source/GameInterface/Services/Clans/Patches/ClanNameChangePatch.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Common;
+using Common.Extensions;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Services.Clans.Messages;
+using HarmonyLib;
+using System;
+using System.Reflection;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Localization;
+
+namespace GameInterface.Services.Clans.Patches
+{
+    [HarmonyPatch(typeof(Clan), nameof(Clan.ChangeClanName))]
+    public class ClanNameChangePatch
+    {
+        private static MethodInfo Clan_Name = typeof(Clan).GetProperty(nameof(Clan.Name)).GetSetMethod(true);
```

Comment:
Arrow function does not store in memory

```suggestion
        private static MethodInfo Clan_Name => typeof(Clan).GetProperty(nameof(Clan.Name)).GetSetMethod(true);
```